src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1065,17 +1065,19 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
 
                                 // initializers can't produce state machines
                                 Debug.Assert((object)initializerStateMachineTypeOpt == null);
-
-                                Debug.Assert(processedInitializers.LoweredInitializers.Kind == BoundKind.StatementList);
                                 Debug.Assert(!hasErrors);
-                                hasErrors = processedInitializers.LoweredInitializers.HasAnyErrors || diagsForCurrentMethod.HasAnyErrors();
+                                hasErrors = lowered.HasAnyErrors || diagsForCurrentMethod.HasAnyErrors();
                                 SetGlobalErrorIfTrue(hasErrors);
 
                                 if (hasErrors)
                                 {
                                     _diagnostics.AddRange(diagsForCurrentMethod);
                                     return;
                                 }
+
+                                // Only do the cast if we haven't returned with some error diagnostics.
```

Comment:
>// Only do the cast if we haven't returned with some error diagnostics. [](start = 32, length = 71)

I think this might cause the compiler do repeat certain work for initializers that wouldn't be done otherwise. I think it would be better to always assign processedInitializers.LoweredInitializers and create the statement list when ```lowered``` is not a BoundStatementList.